rust/exchange_rate/src/lib.rs

```diff
@@ -0,0 +1,231 @@
+use candid::{candid_method, CandidType, Nat, Principal};
+use ic_cdk;
+use ic_cdk_macros::{self, heartbeat, query, update};
+use queues::*;
+use serde::{Deserialize, Serialize};
+use std::cell::RefCell;
+use std::collections::HashMap;
+
+type Timestamp = u64;
+type Rate = f32;
+
+#[derive(CandidType, Clone, Deserialize, Debug, Eq, Hash, PartialEq, Serialize)]
+pub struct HttpHeader {
+    pub name: String,
+    pub value: String,
+}
+
+#[derive(Clone, Debug, PartialEq, CandidType, Eq, Hash, Serialize, Deserialize)]
+pub enum HttpMethod {
+    GET,
+}
+
+#[derive(CandidType, Deserialize, Debug)]
+pub struct CanisterHttpRequestArgs {
+    pub url: String,
+    pub headers: Vec<HttpHeader>,
+    pub body: Option<Vec<u8>>,
+    pub http_method: HttpMethod,
+    pub transform_method_name: Option<String>,
+}
+
+#[derive(CandidType, Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct CanisterHttpResponsePayload {
+    pub status: u64,
+    pub headers: Vec<HttpHeader>,
+    pub body: Vec<u8>,
+}
+
+thread_local! {
+    pub static FETCHED: RefCell<HashMap<Timestamp, Rate>>  = RefCell::new(HashMap::new());
+    pub static REQUESTED: RefCell<Queue<Timestamp>> = RefCell::new(Queue::new());
+
+    pub static RESPONSE_HEADERS_SANTIZATION: Vec<&'static str> = vec![
```

Comment:
Doesn't need to be thread_local AFAIK

Reply:
Right, this static vector doesn't need to be.  Moved to global scope.